admin/preprint_providers/views.py

```diff
@@ -93,15 +92,13 @@ def get_context_data(self, *args, **kwargs):
         for parent in preprint_provider.top_level_subjects:
             subject_html += '<li>{}</li>'.format(parent.text)
             child_html = '<ul>'
-            for child in parent.children.all():
-                if child in all_subjects:
-                    child_html += '<li>{}</li>'.format(child.text)
-                    grandchild_html = '<ul>'
-                    for grandchild in child.children.all():
-                        if grandchild in all_subjects:
-                            grandchild_html += '<li>{}</li>'.format(grandchild.text)
-                    grandchild_html += '</ul>'
-                    child_html += grandchild_html
+            for child in parent.children.filter(id__in=subject_ids):
```

Comment:
If you only need the `text` field, use `.values('text')` or `.values_list('text', flat=true)`.